addons/mass_mailing/models/mass_mailing.py

```diff
@@ -555,14 +604,22 @@ def copy(self, default=None):
     def _group_expand_states(self, states, domain, order):
         return [key for key, val in type(self).state.selection]
 
-    def update_opt_out(self, email, res_ids, value):
-        model = self.env[self.mailing_model_real].with_context(active_test=False)
-        if 'opt_out' in model._fields:
-            email_fname = 'email_from'
-            if 'email' in model._fields:
-                email_fname = 'email'
-            records = model.search([('id', 'in', res_ids), (email_fname, 'ilike', email)])
-            records.write({'opt_out': value})
+    def update_opt_out(self, email, list_ids, value):
+        if len(list_ids) > 0:
+            model = self.env['mail.mass_mailing.contact'].with_context(active_test=False)
+            records = model.search([('email', '=ilike', email)])
+            opt_out_records = self.env['mail.mass_mailing.list_contact_rel']\
```

Comment:
Is the filter necessary ? I mean, something like
```
self.env[].search([
    ('contact_id', 'in', records.ids),
    ('list_id', 'in', list_ids),
    ('opt_out', '!=', value)]
```
cannot work ?

By the way, you can go the a newline inside methods, just avoid \ . Each time you add an \ to go on a new line, a kitten with a lot of allergens comes and try to make you sneeze to death.